File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/utils/AddressUtils.java
package com.example.ama.android2_lesson03.utils;

import android.location.Address;

public class AddressUtils {
    private final static String SEPARATOR = ", ";

    public static String buildFullName(Address address) {
        StringBuilder fullLocationName = new StringBuilder();
        if (address != null) {
            int index = address.getMaxAddressLineIndex();
            for (int i = 0; i <= index; i++) {
                if (fullLocationName.length() != 0) fullLocationName.append(SEPARATOR);
                fullLocationName.append(address.getAddressLine(i));
            }
        }
        return fullLocationName.toString();
    }
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/utils/AddressUtils.kt
package com.example.ama.android2_lesson03.utils

import android.location.Address
import java.lang.StringBuilder

object AddressUtils {

    fun buildFullName(address: Address?): String {
        val builder = StringBuilder()
        if (address != null) {
            for (i in 0..address.maxAddressLineIndex) {
                if (builder.isNotEmpty()) builder.append(",")
                builder.append(address.getAddressLine(i))
            }
        }
        return builder.toString()
    }
}